src/statement.d

```diff
@@ -5149,6 +5154,19 @@ public:
                 }
             }
         }
+
+        if (sc.func)
+        {
+            flags |= sc.func.flags & (FUNCFLAGcppExceptions | FUNCFLAGdExceptions);
+            sc.func.flags |= flags;
+            if (flags == (FUNCFLAGcppExceptions | FUNCFLAGdExceptions))
+            {
+                error("cannot mix catching D and C++ exceptions in the same function");
```

Comment:
Pointer comparison of the TType entries doesn't work, because you can catch the base class of the thrown type. This is true for both D and C++.

The exception class is NOT helpful in determining what types are caught. The only thing that tells what is in the TType is the personality function itself.

We could always invent our own LSDA format, but I'd rather not go that far.